lib/celluloid/io/stream.rb

```diff
@@ -80,6 +71,20 @@ def syswrite(string)
         total_written
       end
 
+      # TODO: remove after ending ruby 2.0.0 support
+      if RUBY_VERSION >= "2.1" 
+        def read_nonblock(*args, **options)
```

Comment:
I'll get back at this tomorrow. But isn't that what this method is doing already? No redefinition is happening under ruby < 2.1, it just delegates to the socket.